library/pkparse.c

```diff
@@ -1306,39 +1307,43 @@ int mbedtls_pk_parse_key( mbedtls_pk_context *pk,
         return( 0 );
 
     mbedtls_pk_free( pk );
+    mbedtls_pk_init( pk );
 
 #if defined(MBEDTLS_RSA_C)
 
     pk_info = mbedtls_pk_info_from_type( MBEDTLS_PK_RSA );
-    if( ( ret = mbedtls_pk_setup( pk, pk_info ) ) != 0 ||
-        ( ret = pk_parse_key_pkcs1_der( mbedtls_pk_rsa( *pk ),
-                                        key, keylen ) ) != 0 )
-    {
-        mbedtls_pk_free( pk );
-    }
-    else
+    if( mbedtls_pk_setup( pk, pk_info ) == 0 &&
+        pk_parse_key_pkcs8_unencrypted_der( mbedtls_pk_rsa( *pk ),
```

Comment:
Is the change from `pkcs1_der` to `pkcs8_encrypted_der` intentional?

Reply:
Gosh... certainly not. Sorry for that.

Reply:
@mpg Do you mind me amending the PR?

Reply:
On the contrary, please amend it!

Reply:
Need to get rid of this commit of shame as quickly as possible...